### DISCUSSION TITLE
Cookie Banner

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#17082] [Feature Requests]
Since the law in germany a website have to show a visitor informations about used cookies. The user must have the option to accept or reject them and to see which cookies are used and has the option to only use essentials cookies.

I hate this topic but since i love directus my question is: Is there someone who implemented a plugin or is it possible to add this as a core "feature"?

**Comment by danielband** [comment]
Hi Nils, I still have not understood how you guys are using Directus at your university, but it seems like you are stretching the boundaries of this app :sweat_smile:. In my view Directus Admin is more like a backend panel for internal users, so most people won´t need a cookie banner. Maybe it could be done via 'Embed' hook though?

btw. can I reach out to you somehow? I am really interested in your work and how you are using Directus.

**Reply by NilsBaumgartner1994** [reply]
Hi Daniel, you can contact me via my website (yesterday created): [https://nilsbaumgartner.de](https://nilsbaumgartner.de)

An Embed hook might be a nice solution.

**Reply by paescuj** [reply]
Yeah, I feel like the new [embed hook](https://docs.directus.io/extensions/hooks.html#embed) would be a perfect fit for such an use case!

However, I wonder whether you need to have a cookie banner for Directus in the first row.
I guess you're relating to the «Cookie-Richtlinie» (Richtlinie 2009/136/EG). The policy states that you need to ask the users for consent when storing cookies which are _not absolutely necessary_.
As far as I know, the only cookie stored by Directus is the refresh token and this one is essential for authentication (which in turn is clearly needed 😄).

**Reply by NilsBaumgartner1994** [reply]
> Yeah, I feel like the new [embed hook](https://docs.directus.io/extensions/hooks.html#embed) would be a perfect fit for such an use case!
> 
> However, I wonder whether you need to have a cookie banner for Directus in the first row. I guess you're relating to the «Cookie-Richtlinie» (Richtlinie 2009/136/EG). The policy states that you need to ask the users for consent when storing cookies which are _not absolutely necessary_. As far as I know, the only cookie stored by Directus is the refresh token and this one is essential for authentication (which in turn is clearly needed 😄).

Yes you are right, essential cookies are not needed to show if no other cookies are used

I thought about the use case for using Directus as blog post website. In my head was the part that you always would need a banner.

**Reply by paescuj** [reply]
> In my head was the part that you always would need a banner.

Fair enough, those banners are omnipresent! 🤣🙈

> I thought about the use case for using Directus as blog post website.

In that case, I think Directus will serve as the backend and you could set-up a cookie banner in your frontend (maybe even using an existing library like https://www.npmjs.com/package/@beyonk/gdpr-cookie-consent-banner).

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
